unix/network.ml

```diff
@@ -89,13 +89,19 @@ let connect net ~sw ~secret_key (addr, auth) =
         let socket = Eio_unix.Resource.fd_opt socket |> Option.get in
         Eio_unix.Fd.use_exn "keep-alive" socket @@ fun socket ->
         Unix.setsockopt socket Unix.SO_KEEPALIVE true;
+        Unix.setsockopt socket Unix.TCP_NODELAY true;
         Keepalive.try_set_idle socket 60
     end;
     Tls_wrapper.connect_as_client socket secret_key auth
   | exception ex ->
     error "@[<v2>Network connection for %a failed:@,%a@]" Location.pp addr Fmt.exn ex
 
 let accept_connection ~secret_key flow =
+  Eio_unix.Resource.fd_opt flow
+  |> Option.iter (fun fd ->
+      Eio_unix.Fd.use_exn "TCP_NODELAY" fd @@ fun fd ->
+      Unix.setsockopt fd Unix.TCP_NODELAY true
```

Comment:
We might need to wrap this to catch

```ocaml
Unix.Unix_error(Unix.EOPNOTSUPP, "setsockopt", "")
```

Stumbled across this new change when using this branch on macOS with a unix domain socket.